### PR TITLE
PR: Crashing on shortcut assignment with PyQt5

### PR DESCRIPTION
**Pull request by Qlogin**
fix crash on shortcut assignment (Qt5 'dataChanged' signal has 3 parameters)

### CODE REVIEW

File: spyderlib/plugins/shortcuts.py [hunk 1]
```diff
@@ -789,8 +789,9 @@ def setup_page(self):
         self.setTabOrder(self.finder, self.reset_btn)
 
         # Signals and slots
+        #    Qt5 'dataChanged' has 3 parameters
         self.table.proxy_model.dataChanged.connect(
-                     lambda i1, i2, opt='': self.has_been_modified(opt))
+                     lambda i1, i2, roles=[], opt='': self.has_been_modified(opt))
         self.reset_btn.clicked.connect(self.reset_to_default)
```
Comment (goanpeca) [inline spyderlib/plugins/shortcuts.py]: Hi @Qlogin, thanks for working on this.

I think that to make the code much clearer to future contributors we could check the current API in use and then feed the correct connection

Like

```python
from spyderlib.qt import PYQT5

...

if PYQT5:
    #    Qt5 'dataChanged' has 3 parameters
    self.table.proxy_model.dataChanged.connect(lambda i1, i2, roles=[], opt='': self.has_been_modified(opt))
else:
    self.table.proxy_model.dataChanged.connect(lambda i1, i2, opt='': self.has_been_modified(opt))

**Comment by ccordoba12** [comment]
I agree with @goanpeca too.

**Comment by goanpeca** [comment]
@Qlogin any updates?

**Comment by Qlogin** [comment]
Sorry. Problems with e-mail notifications.
Updates have been committed

**Comment by goanpeca** [comment]
Neat!, thanks, this is good to go, right @ccordoba12 ?

**Comment by goanpeca** [comment]
@ccordoba12 ?

**Comment by ccordoba12** [comment]
@Qlogin, thanks a lot for your contribution and sorry for the big delay in merging :-)

**Pull request merged**